### ISSUE TITLE
AcquireTokenSilent sometimes ignores tenant constraint

### ISSUE THREAD

**Issue by bgavrilMS** [AzureAD/microsoft-authentication-library-for-dotnet#1123]
**Which Version of MSAL are you using ?**
Note that to get help, you need to run the latest version. Preview version are also ok. 
For ADAL, please log issues to https://github.com/AzureAD/azure-activedirectory-library-for-dotnet
3.0.6

**Platform**
all

Other? - please describe;

**Repro**

```csharp
// create PCA with defaults
// Login a user using **tenant1**

var account = await pca.GetAccountsAsync();
var result = await pca
                     .AcquireTokenSilentAsunc(Scopes, account)
                     .WithAuthority("https://login.microsoftonline.com/<**tenant2**>")
                     .ExecuteAsync();

```

**Expected behavior**
no result / exception

**Actual behavior**
token for tenant1 is returned

**Comment by bgavrilMS** [comment]
https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/pull/1122

**Comment by jennyf19** [comment]
fixed in [3.0.8](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/releases/tag/3.0.8)
